pkg/controller/controllerutil/controllerutil.go

```diff
@@ -171,7 +171,27 @@ func RemoveControllerReference(owner, object metav1.Object, scheme *runtime.Sche
 	if ok := HasControllerReference(object); !ok {
 		return fmt.Errorf("%T does not have a owner reference with controller equals true", object)
 	}
-	return RemoveOwnerReference(owner, object, scheme)
+	ro, ok := owner.(runtime.Object)
+	if !ok {
+		return fmt.Errorf("%T is not a runtime.Object, cannot call RemoveControllerReference", owner)
+	}
+	gvk, err := apiutil.GVKForObject(ro, scheme)
+	if err != nil {
+		return err
+	}
+	ownerRefs := object.GetOwnerReferences()
+	index := indexOwnerRef(ownerRefs, metav1.OwnerReference{
+		APIVersion: gvk.GroupVersion().String(),
+		Name:       owner.GetName(),
+		Kind:       gvk.Kind,
+		Controller: ptr.To(true),
+	})
+	if index == -1 {
+		return fmt.Errorf("%T does not have an controller reference for %T", object, owner)
+	}
+	ownerRefs = append(ownerRefs[:index], ownerRefs[index+1:]...)
+	object.SetOwnerReferences(ownerRefs)
+	return nil
```

Comment:
Not a fan of this replicated logic but didn't want to assert it to a `runtime.Object` back to a `metav1.Object` to utilize the reuse.  

This seems to make sure the only controllerReference is the owner that is passed in because 
1. Only one ControllerReference (1:1) be present on an object but many OwnerReferences (1:N)
2. The index returned verifies that the above owner is in fact the controller reference on the object if the index is greater than -1.